R/sprintf_linter.R

```diff
@@ -21,51 +21,52 @@ sprintf_linter <- function() {
 
     sprintf_calls <- xml2::xml_find_all(xml, xpath)
 
-    get_range_text <- function(content, line1, col1, line2, col2) {
-      lines <- content[line1:line2]
-      lines[length(lines)] <- substr(lines[length(lines)], 1L, col2)
-      lines[1L] <- substr(lines[1L], col1, nchar(lines[1L]))
-      lines
-    }
-
-    line1 <- as.integer(xml2::xml_attr(sprintf_calls, "line1"))
-    col1 <- as.integer(xml2::xml_attr(sprintf_calls, "col1"))
-    line2 <- as.integer(xml2::xml_attr(sprintf_calls, "line2"))
-    col2 <- as.integer(xml2::xml_attr(sprintf_calls, "col2"))
+    message <- vapply(sprintf_calls, function(xml) {
```

Comment:
this is quite a long inline function, at a glance I don't see it inheriting from the function environment -- let's make it its own helper?